rir/src/compiler/pir/instruction.h

```diff
@@ -790,6 +790,36 @@ class FLIE(Subassign2_1D, 4, Effect::None, EnvAccess::Leak) {
     Value* idx() { return arg(2).val(); }
 };
 
+class FLIE(Subassign1_2D, 5, Effect::None, EnvAccess::Leak) {
+  public:
+    Subassign1_2D(Value* val, Value* vec, Value* idx1, Value* idx2, Value* env,
+                  unsigned srcIdx)
+        : FixedLenInstructionWithEnvSlot(PirType::val(),
+                                         {{PirType::val(), PirType::val(),
+                                           PirType::val(), PirType::val()}},
+                                         {{val, vec, idx1, idx2}}, env,
+                                         srcIdx) {}
+    Value* rhs() { return arg(0).val(); }
+    Value* lhsValue() { return arg(1).val(); }
```

Comment:
I would rather say `rhsValue` and `lhs`, because the left side is the matrix you are subassigning, right?

Reply:
I changed both